sktime/transformations/series/impute.py

```diff
@@ -111,19 +111,21 @@ def transform(self, Z, X=None):
                 forecaster = self.forecaster
             else:
                 forecaster = PolynomialTrendForecaster(degree=1)
-            # in-sample forecasting horizon
-            fh_ins = -np.arange(len(Z))
             # fill NaN before fitting with ffill and backfill (heuristic)
-            Z = Z.fillna(method="ffill").fillna(method="backfill")
+            Z_aux = Z.fillna(method="ffill").fillna(method="backfill")
             # multivariate
-            if isinstance(Z, pd.DataFrame):
-                for col in Z:
-                    forecaster.fit(y=Z[col])
+            if isinstance(Z_aux, pd.DataFrame):
+                for col in Z_aux:
+                    forecaster.fit(y=Z_aux[col])
+                    # in-sample forecasting horizon
+                    fh_ins = np.where(Z[col].isna())[0] - (len(Z)-1)
```

Comment:
Can we not simplify this when using the `ForecastingHorizon` class from `sktime.forecasting.base`?

Reply:
I didn't understange what you suggested exactly. Is it something like this?

```python
            # fill NaN before fitting with ffill and backfill (heuristic)
            Z_aux = Z.fillna(method="ffill").fillna(method="backfill")
            # multivariate
            if isinstance(Z_aux, pd.DataFrame):
                for col in Z_aux:
                    forecaster.fit(y=Z_aux[col])
                    # in-sample forecasting horizon
                    fh_ins = ForecastingHorizon(values=np.where(Z[col].isna())[0], is_relative=False)
                    fh_ins = fh_ins.to_relative(cutoff=len(Z)-1).to_numpy()
                    Z_pred = forecaster.predict(fh=fh_ins)
                    Z[col] = Z[col].fillna(value=Z_pred)
            # univariate
            else:
                forecaster.fit(y=Z_aux)
                # in-sample forecasting horizon
                fh_ins = ForecastingHorizon(values=np.where(Z.isna())[0], is_relative=False)
                fh_ins = fh_ins.to_relative(cutoff=len(Z)-1).to_numpy()
                Z_pred = forecaster.predict(fh=fh_ins)
                Z = Z.fillna(value=Z_pred)
```

Reply:
Sorry for the slow reply @julramos, I think you can simplify it even further, here's a simple example:
```python
y = load_airline()
y.iloc[np.random.randint(0, len(y), 20)] = np.nan

forecaster = AutoARIMA(d=0, suppress_warnings=True)

isna = y.isna()
fh = ForecastingHorizon(y.index[isna], is_relative=False)  # define fh based on index of missing values 
y_aux = y.fillna(method="ffill").fillna(method="backfill")
forecaster.fit(y_aux, fh=fh) 
y_pred = forecaster.predict()
y.iloc[isna] = y_pred  # replace missing values with predicted values
```

A few remarks:
* make sure to pass `fh` to fit rather than predict (some forecaster need the horizon already during fitting)
* in the above if-else statement there seems to be some code duplication, could we write a function for univariate imputation based on the forecaster and then loop over it with the column selection?